### PR TITLE
Draft: migrate to Cats Effect 3

### PR DESCRIPTION
**Pull request by DeviLab**

### CODE REVIEW

File: skafka/src/main/scala/com/evolutiongaming/skafka/producer/Producer.scala [hunk 1]
```diff
@@ -113,15 +112,14 @@ object Producer {
     fromProducerJ1(producer)
   }
 
-
   private sealed abstract class Main
 
   @deprecated("use `fromProducerJ1` instead", "11.7.0")
-  def fromProducerJ[F[_]: Effect: Blocking](producer: F[ProducerJ[Bytes, Bytes]]): Resource[F, Producer[F]] = {
+  def fromProducerJ[F[_]: Blocking: Sync: Async](producer: F[ProducerJ[Bytes, Bytes]]): Resource[F, Producer[F]] = {
```
Comment (t3hnar) [inline skafka/src/main/scala/com/evolutiongaming/skafka/producer/Producer.scala:118]: do we need both `Sync` & `Async` separately here?

Comment (t3hnar) [reply skafka/src/main/scala/com/evolutiongaming/skafka/producer/Producer.scala:118]: I think we should delete deprecated methods and not try to migrate them to CE3

**Review by t3hnar — CHANGES_REQUESTED** [review]

File: skafka/src/main/scala/com/evolutiongaming/skafka/producer/Producer.scala (hunk 1 above)
Comment (DeviLab) [reply skafka/src/main/scala/com/evolutiongaming/skafka/producer/Producer.scala]: Removed deprecated methods

**Comment by t3hnar** [comment]
overall looks ok, now we need dependencies in place prior merging this :)

**Pull request merged**